app/src/main/java/com/ably/tracking/demo/publisher/KoinModule.kt

```diff
@@ -53,5 +62,23 @@ val appModule = module {
 
     single { NotificationProvider(get()) }
 
+    factory { buildOkHttpClient() }
+
+    factory { buildRetrofit(get(), BuildConfig.FIREBASE_REGION, BuildConfig.FIREBASE_PROJECT_NAME) }
+
+    factory { buildDeliveryServiceApi(get()) }
+
+    factory { RetrofitDeliveryServiceApiSource(get()) } bind DeliveryServiceApiSource::class
```

Comment:
Why all of them are `factory` and not `single`? :thinking: Do we want to create an entirely new instance of `Retrofit` and other services for each component that asks for it? :thinking:

Reply:
https://github.com/ably/asset-tracking-android-rider-app-demo/pull/39/commits/bdafac00ef69089740b19cdd4d63ca5732750780